src/metric/elementwise_metric.cu

```diff
@@ -346,7 +346,10 @@ struct EvalEWiseBase : public Metric {
         << "hint: use merror or mlogloss for multi-class classification";
     const auto ndata = static_cast<omp_ulong>(info.labels_.Size());
     // Dealing with ndata < n_gpus.
-    GPUSet devices = GPUSet::All(tparam_->gpu_id, tparam_->n_gpus, ndata);
+    // Perform the metrics computation on CPU if device list isn't present in prediction vector
```

Comment:
I think there can a better heuristic on this configuration.  Please try to document the proposed behavior.